stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/list/Messages.kt

```diff
@@ -181,65 +208,160 @@ private fun OnLastVisibleItemChanged(lazyListState: LazyListState, onChanged: (f
  * Represents the default scrolling behavior and UI for [Messages], based on the state of messages and the scroll state.
  *
  * @param messagesState The state of messages, current message list, thread, user and more.
- * @param lazyListState The scrolling state of the list, used to manipulate and trigger scroll events.
+ * @param messagesLazyListState The scrolling state of the list, used to manipulate and trigger scroll events.
+ * @param scrollToBottom Handler when the user requests to scroll to the bottom of the messages list.
  */
 @Composable
 internal fun BoxScope.DefaultMessagesHelperContent(
     messagesState: MessagesState,
-    lazyListState: LazyListState,
+    messagesLazyListState: MessagesLazyListState,
+    scrollToBottom: (() -> Unit) -> Unit,
 ) {
-    val (_, _, _, messages, _, _, newMessageState) = messagesState
+    val lazyListState = messagesLazyListState.lazyListState
+
+    val messages = messagesState.messageItems
+    val newMessageState = messagesState.newMessageState
+    val areNewestMessagesLoaded = messagesState.newestMessageLoaded
+    val isMessageInThread = messagesState.parentMessageId != null
+
     val coroutineScope = rememberCoroutineScope()
 
     val firstVisibleItemIndex = lazyListState.firstVisibleItemIndex
```

Comment:
I see we removed the switch to derived state here.
I know the linter wants this to be derived so I'm wondering why we reverted.

Reply:
added derived back in. Removed it when fixing a bug reported by Dmitrii, had to remove remember since it was rememberins an old instance of lazyListState and was causing bugs.